apps/documents/views.py

```diff
@@ -0,0 +1,231 @@
+@require_POST
+@permission_required("files.change_file")
+def edit_file(request, team_slug: str, id: int):
+    file = get_object_or_404(File.objects.defer("file"), team__slug=team_slug, id=id)
+    file.name = request.POST.get("name")
+    file.summary = request.POST.get("summary")
+
+    existing_collections = set(
+        file.repository_set.filter(type=RepositoryType.COLLECTION).values_list("name", flat=True)
+    )
+    collection_set = set()
+    # Handle new collections
+    for collection_name in request.POST.getlist("collections[]"):
+        collection_set.add(collection_name)
+        repo = get_object_or_404(Repository, team__slug=team_slug, type=RepositoryType.COLLECTION, name=collection_name)
+        repo.files.add(file)
+    file.save(update_fields=["name", "summary"])
+
+    # Remove from collections
+    for collection_name in existing_collections - collection_set:
+        repo = get_object_or_404(Repository, team__slug=team_slug, type=RepositoryType.COLLECTION, name=collection_name)
+        repo.files.remove(file)
+
+    return redirect(reverse("documents:repositories", kwargs={"team_slug": team_slug, "tab_name": "files"}))
+
+
+class CollectionListView(LoginAndTeamRequiredMixin, BaseObjectListView):
+    template_name = "documents/shared/list.html"
+    model = Repository
+    paginate_by = 10
+    details_url_name = "documents:collection_details"
+    tab_name = "collections"
+    permission_required = "documents.view_repository"
+
+    def get_queryset(self):
+        queryset = (
+            super()
+            .get_queryset()
+            .filter(type=RepositoryType.COLLECTION, team__slug=self.kwargs["team_slug"])
+            .order_by("-created_at")
+        )
+
+        search = self.request.GET.get("search")
+        if search:
+            name_similarity = TrigramSimilarity("name", search)
+            queryset = (
+                queryset.annotate(similarity=name_similarity).filter(Q(similarity__gt=0.2)).order_by("-similarity")
+            )
+        return queryset
+
+
+class CollectionDetails(LoginAndTeamRequiredMixin, BaseDetailsView):
+    template_name = "documents/collection_details.html"
+    model = Repository
+    permission_required = "documents.view_repository"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        collection = self.get_object()
+        context["edit_url"] = reverse(
+            "documents:edit_collection", kwargs={"team_slug": self.kwargs["team_slug"], "id": collection.id}
+        )
+        context["delete_url"] = reverse(
+            "documents:delete_collection", kwargs={"team_slug": self.kwargs["team_slug"], "id": collection.id}
+        )
+        return context
+
+
+@require_POST
+@transaction.atomic()
+def new_collection(request, team_slug: str):
+    """Create a new collection"""
+    Repository.objects.create(type=RepositoryType.COLLECTION, team=request.team, name=request.POST.get("name"))
```

Comment:
This needs to handle uniqueness failures with a useful error message.

Reply:
https://github.com/dimagi/open-chat-studio/pull/1259/commits/7d8614b273f9666ce10f429e43ce97d2fbcbbe7c